### ISSUE TITLE
(Pic Included) What does 'labels' signify on the terminal console during training?

### ISSUE THREAD

**Issue by BuilderBoba** [ultralytics/yolov5#3801]
## Additional context
![labels2](https://user-images.githubusercontent.com/28004118/123568599-a03e4580-d7ff-11eb-8353-47505fb3af91.JPG)
I'm having difficulty understanding what the numbers below 'labels' signify and why it fluctuates between epochs.
For information about my training set, I have 501 labelled pictures as my training data (nc=2) running at 150 epochs.

So to wrap up,
1. What is 'labels' showing or representing here?
2. Why does it change between epochs?

Additional Q
3. If img_size is set to 640 like the picture above, does that mean the image resolution for my images as input to the training model is always resized to (640x640)? So if I have for example images of resolutions (1280x500) and (120x700) they will be resized to 640x640 when they are used as training?

**Comment by glenn-jocher** [comment]
@BuilderBoba labels per batch.

**Comment by BuilderBoba** [comment]
@glenn-jocher Hi Glenn, thank you for your response.

Could you elaborate on why the labels per batch number is shown as above, and if there is a certain equation towards calculating that value?

What I mean is, I had my training model running at batch_size 16, and I'm having trouble grasping why a 16 batch_size would only have 9 or 11 labels per batch. Shouldn't the labels per batch be 16 except for the residual batch? Also, the fluctuation of the label values seems erratic -- they don't just oscillate between 9 and 11 exclusively, but sometimes the values are different, like 8 or 13.  Why doesn't the metric show a more consistent output of numbers per epoch? 

I would really appreciate a thorough response since I am currently writing a tutorial documentation for Korean readers, and your input can greatly assist the Korean AI Community in learning this great tool!

**Comment by glenn-jocher** [comment]
@BuilderBoba a label is a box.

**Comment by glenn-jocher** [comment]
@BuilderBoba you can see the number of labels in each batch by looking at your train_batch.jpg images, i.e. this is the first batch, train_batch0.jpg with 16 mosaics:

![train_batch0](https://user-images.githubusercontent.com/26833433/124086872-c4916080-da51-11eb-8c7a-5b21bc87ed55.jpg)
